lib/docker-toolbelt.coffee

```diff
@@ -107,12 +112,49 @@ EEXIST = code: 'EEXIST'
 ignore = ->
 MIN_PAGE_SIZE = 4096
 
+pathPrefixRemover = (prefix) ->
+	(path) ->
+		slice = path.substr(prefix.length)
+		# return original if path doesn't start with given prefix
+		return if "#{prefix}#{slice}" == path then slice else path
+
+overlay2MountWithDisposer = (fsRoot, target, lowers, diffDir, workDir) ->
+	# If no lower, just return diff directory
+	return Promise.resolve(diffDir) if !lowers
+
+	fs.mkdirAsync(target)
+	.catch EEXIST, ignore
+	.then ->
+		options = "lowerdir=#{lowers},upperdir=#{diffDir},workdir=#{workDir}"
+		mountOpts = {}
+		# Use relative paths when the mount data has exceeded the page size.
+		# The mount syscall fails if the mount data cannot fit within a page and
+		# relative links make the mount data much smaller.
+		if options.length > MIN_PAGE_SIZE
+			mountOpts.cwd = fsRoot
+			makeRelative = pathPrefixRemover(fsRoot + '/')
```

Comment:
It would be better to use
```coffee
path = require('path')
...
makeRelative = pathPrefixRemover(fsRoot + path.sep)
```
to avoid hardcoding the path separator